lib/matplotlib/axes/_axes.py

```diff
@@ -1200,15 +1202,17 @@ def vlines(self, x, ymin, ymax, colors='k', linestyles='solid',
         if not np.iterable(ymax):
             ymax = [ymax]
 
-        x, ymin, ymax = cbook.delete_masked_points(x, ymin, ymax)
-
+        # Create and combine masked_arrays from input
+        x, ymin, ymax = cbook._combine_masks(x, ymin, ymax)
         x = np.ravel(x)
-        ymin = np.resize(ymin, x.shape)
-        ymax = np.resize(ymax, x.shape)
-
-        verts = [((thisx, thisymin), (thisx, thisymax))
-                 for thisx, thisymin, thisymax in zip(x, ymin, ymax)]
-        lines = mcoll.LineCollection(verts, colors=colors,
+        ymin = np.ma.resize(ymin, x.shape)
+        ymax = np.ma.resize(ymax, x.shape)
+
+        masked_verts = [np.ma.array([xymin, xymax])
```

Comment:
Simpler and faster:
~~~
        masked_verts = np.ma.empty((len(x), 2, 2))
        masked_verts[:, 0, 0] = x
        masked_verts[:, 1, 0] = x
        masked_verts[:, 0, 1] = ymin
        masked_verts[:, 1, 1] = ymax
~~~

Reply:
Yes, that is definitely cleaner than what I had before, thank you for the suggestion! Pushed.